src/main/java/uk/gov/justice/digital/domains/DomainExecutor.java

```diff
@@ -0,0 +1,248 @@
+                        spark.catalog().dropTempView(source);
+                    }
+                }
+            } catch (Exception e) {
+                // TODO handle errors
+                // continue;
+            }
+        }
+    }
+
+
+    // TODO: Mapping will be enhanced at later stage in future user stories
+    protected Dataset<Row> applyMappings(final Dataset<Row> dataFrame, final TableDefinition.MappingDefinition mapping) {
+        if(mapping != null && mapping.getViewText() != null && !mapping.getViewText().isEmpty()) {
+            return dataFrame.sqlContext().sql(mapping.getViewText()).toDF();
+        }
+        return dataFrame;
+    }
+
+    protected Dataset<Row> getAllSourcesForTable(final String source, final TableTuple exclude) {
+        if(exclude != null && exclude.asString().equalsIgnoreCase(source)) {
+            //TODO: this condition only for unit test
+            // we already have this table
+            logger.info("table already present " + exclude.asString());
+        } else {
+            try {
+                TableTuple full = new TableTuple(source);
+                final Dataset<Row> dataFrame = deltaService.load(sourceRootPath, full.getSchema(), full.getTable());
+                if(dataFrame != null) {
+                    logger.info("Loaded source '" + full.asString() +"'.");
+                    return dataFrame;
+                }
+            } catch(Exception e) {
+                handleError(e);
+            }
+        }
+        return null;
+    }
+
+    public Dataset<Row> apply(final TableDefinition table, final Map<String, Dataset<Row>> sourceTableMap)
+            throws DomainExecutorException {
+        try {
+
+            logger.info("DomainExecutor::applyTransform(" + table.getName() + ")...");
+            // Transform
+            Map<String,Dataset<Row>> refs = new HashMap<>();
+            // Add sourceTable if present
+            if (sourceTableMap != null && sourceTableMap.size() > 0) {
+                refs.putAll(sourceTableMap);
+            } else if(table.getTransform() != null && table.getTransform().getSources() != null
+                    && table.getTransform().getSources().size() > 0) {
+                for (final String source : table.getTransform().getSources()) {
+                    Dataset<Row> sourceDataFrame = this.getAllSourcesForTable(source, null);
+                    if (sourceDataFrame != null) {
+                        refs.put(source.toLowerCase(), sourceDataFrame);
+                    } else {
+                        System.err.println("Unable to load source '" + source +"' for Table Definition '" + table.getName() + "'");
+                        throw new DomainExecutorException("Unable to load source '" + source +
+                                "' for Table Definition '" + table.getName() + "'");
+                    }
+                }
+            } else {
+                //TODO: Expecting this condition should never be reached
```

Comment:
Consider removing this entirely if it's never going to happen. If it could happen and it's an issue then we should take the appropriate course of action here (e.g. throw an exception if it's an error that will stop processing from continuing).